### PR TITLE
fix: firstItem-related issues

### PR DESCRIPTION
**Pull request by KingMario**
Signed-off-by: KingMario <gcyyq@hotmail.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [x] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [x] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?

The value description of firstItem and lastItem is not complete.

The offset for API call is not accurate for server-driven datagrid.

Issue Number: N/A

## What is the new behavior?

The desciption is updated.

The offset is updated for server-driven datagrid.

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

### CODE REVIEW

File: src/clr-angular/data/datagrid/providers/state.provider.ts [hunk 1]
```diff
@@ -40,11 +40,14 @@ export class StateProvider<T> {
     const state: ClrDatagridStateInterface<T> = {};
     if (this.page.size > 0) {
       state.page = {
-        from: this.page.firstItem,
-        to: this.page.lastItem,
         size: this.page.size,
         current: this.page.current,
       };
+
+      if (this.page.firstItem !== -1) {
```
Comment (bdryanovski) [inline src/clr-angular/data/datagrid/providers/state.provider.ts:47]: I think that 
```ts
if (this.page.firstItem > 0) {
```

This way .. we won't have to handle `-2` :)

**Comment by KingMario** [comment]
@bdryanovski
Thanks.

The if statement is updated to
```
if (this.page.firstItem >= 0) {
```

unit test for the page info of the state has been added.

**Comment by bdryanovski** [comment]
Hey @KingMario, thanks for the quick response and for the unit-test. I didn't even notice that there is no unit-test, after all, to start off.

**Comment by KingMario** [comment]
@Jinnie @bdryanovski I've modified the doc and done some bug fixes in the demo and website apps. For the state provider, things are kept as they are. And the new-added partial unit test is left to start off.

Please review it again. Thanks.

**Comment by Jinnie** [comment]
@KingMario Sorry for the delay, we've not forgotten this PR, but working on higher priority tasks. I hope to be able to address it early next week.

**Comment by Shijir** [comment]
On top of what Ivan said, there were quite a few times this "issue" had been brought up. There are reasons we haven't addressed it because [this comment/explanation](https://github.com/vmware/clarity/pull/526#pullrequestreview-23768394) still applies today.

**Comment by KingMario** [comment]
@Jinnie Thanks.

I'll add a new PR for the v3 branch.

**Comment by Jinnie** [comment]
Closing in favor of #4608

**Pull request closed without merging**